ssh: T8090: T8098: add support for config test mode (sshd -t)

c-po opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change summary
<!--- Provide a general summary of your changes in the Title above -->

Add a safety-net for development to check if the rendered sshd(8) configuration can be applied at all. If it can't be applied - throw an error.

From https://linux.die.net/man/8/sshd:
```
  -t    Test mode. Only check the validity of the configuration file and sanity
        of the keys. This is useful for updating sshd reliably as configuration
        options may change.
```
Explicitly forcing a broken config now results in:

```
vyos@vyos# commit
[ service ssh ]
Unexpected error with SSH configuration! /run/sshd/sshd_config line 21:
X11DisplayOffset integer value invalid.
[[service ssh]] failed
Commit failed
```

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
* https://vyos.dev/T8090

## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->
* https://github.com/vyos/vyos-documentation/pull/1722

## How to test / Smoketest result

```
vyos@vyos:~$ /usr/libexec/vyos/tests/smoke/cli/test_service_ssh.py
test_ssh_default (__main__.TestServiceSSH.test_ssh_default) ... ok
test_ssh_dynamic_protection (__main__.TestServiceSSH.test_ssh_dynamic_protection) ... ok
test_ssh_login (__main__.TestServiceSSH.test_ssh_login) ... ok
test_ssh_multiple_listen_addresses (__main__.TestServiceSSH.test_ssh_multiple_listen_addresses) ... ok
test_ssh_ndcpp (__main__.TestServiceSSH.test_ssh_ndcpp) ... ok
test_ssh_pubkey_accepted_algorithm (__main__.TestServiceSSH.test_ssh_pubkey_accepted_algorithm) ... ok
test_ssh_single_listen_address (__main__.TestServiceSSH.test_ssh_single_listen_address) ... ok
test_ssh_trusted_user_ca (__main__.TestServiceSSH.test_ssh_trusted_user_ca) ... ok
test_ssh_vrf_multi (__main__.TestServiceSSH.test_ssh_vrf_multi) ... ok
test_ssh_vrf_single (__main__.TestServiceSSH.test_ssh_vrf_single) ... ok

----------------------------------------------------------------------
Ran 10 tests in 47.068s

OK
```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

sever-sever reviewed (approved):
Add sshd test option, to check config in the verify and provide better information aobut errors.

natali-rs1985 reviewed (commented):
Config load test failed 
```
DEBUG - [ service ssh ]
DEBUG - Unexpected error with SSH configuration! /run/sshd/sshd_config line 55:
DEBUG - Bad SSH2 cipher spec 'aes128-ctr,aes192-ctr,aes256-ctr,chacha20-
DEBUG - poly1305@openssh.com,rijndael-cbc@lysator.liu.se'.
DEBUG - [[service ssh]] failed
DEBUG - Commit failed
```
https://github.com/vyos/vyos-1x/actions/runs/20137711221/job/57797536733#step:5:1766

c-po commented:
> Config load test failed

Very nice - our smoketests found a hidden bug https://vyos.dev/T8098

c-po committed:
ssh: T8090: add support for config test mode (sshd -t)

Add a safety-net for development to check if the rendered sshd(8) configuration
can be applied at all. If it can't be applied - throw an error.

From https://linux.die.net/man/8/sshd:
  -t    Test mode. Only check the validity of the configuration file and sanity
        of the keys. This is useful for updating sshd reliably as configuration
        options may change.

Explicitly forcing a broken config now results in:

  vyos@vyos# commit
  [ service ssh ]
  Unexpected error with SSH configuration! /run/sshd/sshd_config line 21:
  X11DisplayOffset integer value invalid.
  [[service ssh]] failed
  Commit failed

c-po committed:
ssh: T8098: remove support for deprecated "rijndael-cbc@lysator.liu.se" cipher

According to an Arch Linux forum discussion, the cipher
rijndael-cbc@lysator.liu.se was removed in OpenSSH 6.7.

References:
- https://bbs.archlinux.org/viewtopic.php?id=188613
- https://www.openssh.org/txt/release-6.7
- https://github.com/openssh/openssh-portable/commit/03e93c753d7c223063a

c-po committed:
ssh: T8098: rename "ciphers" CLI node to "cipher"

Follow VyOS CLI best practices for using singular whenever possible to build a
CLI node. As we introduce a new migration 2 -> 3 for SSH we can correct this
minor detail.

c-po commented:
`sagitta` will receive it's own fix where the selected SSH cipher is simply not rendered to not break any automation in the wild.
It is properly migrated away once system is upgraded to VyOS 1.5

This will be done by highjacking the linked backport PR for `sagitta` so everything is still properly linked.

dmbaturin reviewed (approved):
The change from `ciphers` to `cipher` is in line with our convention that node names should be singular nouns; the idea to run `sshd -t` as part of `verify()` is very good. I don't have any objections.
